### ISSUE TITLE
ErrorControlChain.php turns display_errors = Off

### ISSUE THREAD

**Issue by Zauberfisch** [silverstripe/silverstripe-framework#2272]
SilverStripe version: latest 3.1 (d44024b1cf642bc47753c73341139fd5790a2a76)

ErrorControlChain set display_errors to 0
it does store the old value, and there is some code that seems to be suppose to restore the old value, but this is not happening (or maybe not in time)

my _config.php contains:

    ...
    SS_Log::add_writer(new SS_LogFileWriter('../error.log'), SS_Log::ERR);
    ini_set('display_errors', 1);
    error_reporting(E_ALL);

lets say we have an syntax error in Page_Controller->init() (like missing a ; somewhere)
the result is a blank page and only the apache2 error log shows the actually error message.
(but Exceptions and user_errors do work as far as I can tell)

I consider this bug a blocker, as it will greatly effect many developers.
printing a phpinfo() in _config.php will show that display_errors = on, printing a phpinfo() in Page_Controller->init() will show display_errors = off

**Comment by shoaibali** [comment]
Interesting this is still open... I am running version 3.1.2 main.php also bypasses suppression, this has been fixed in more recent versions by checking for redirects. Time to upgrade, not seeing errors was getting very annoying.

```php

		// First, if $_GET['flush'] was set, but no valid token, suppress the flush
		if (isset($_GET['flush']) && !$token->tokenProvided()) {
			unset($_GET['flush']);
		}
		else {
			$chain->setSuppression(false);
		}

```

**Comment by dhensby** [comment]
fixed by #5555 I believe
